Utils files for general functions

Raphael-Leb opened the pull request:
Closes #84

Raphael-Leb committed:
Add utils for general functions #84

Move the following function to a new utils class:

is_market_open()

Add a new function to the utils class:

A function that gets the date that is x amount of market day(s) away

Raphael-Leb committed:
#87 Added utils.py

TristanFecteau committed:
Revert "#87 Added utils.py"

This reverts commit b4aa38699e62be368753651e1f58640d08306f40.

TristanFecteau commented:
1. I changed the name of the pull request since it didn't mean anything
2. I reverted one of your commits since it was just copying all of our changes on develop

TristanFecteau commented on src/bullets/utils/market_open_utils.py:
```diff
@@ -0,0 +1,26 @@
+
+        return date not in us_holiday_list(date.year)
+
+def get_date_x_days_away(x):
```
x is not a valid name for a parameter
x should have a type specified
get_date_x_days_away should have a return type specified
the function has no documentation

TristanFecteau commented on src/bullets/utils/market_open_utils.py:
```diff
@@ -0,0 +1,26 @@
+from bullets.data_source.data_source_interface import Resolution
+from bullets.utils.holiday_date_util import us_holiday_list
+
+def is_market_open(date: datetime, resolution: Resolution=Resolution.DAILY) -> bool:
```
the function has no documentation

TristanFecteau reviewed (changes requested):
See files changed comments for line-specific changes to make

Raphael-Leb committed:
Revision of #84

Added description, changed variables names and imposed argument and return types

Raphael-Leb committed:
Merge branch 'Do-it-here' of https://github.com/AlgoETS/BullETS into Do-it-here

Raphael-Leb committed:
Update .gitignore

Raphael-Leb committed:
Update .gitignore

Raphael-Leb committed:
Second revision of #84

Added documentation for is_market_open()

Raphael-Leb committed:
Merge branch 'Do-it-here' of https://github.com/AlgoETS/BullETS into Do-it-here

Raphael-Leb committed:
Update .gitignore

Raphael-Leb committed:
Corrected indentation error on #84

Raphael-Leb committed:
Merge branch 'Do-it-here' of https://github.com/AlgoETS/BullETS into Do-it-here

TristanFecteau committed:
changed file name & reformatted file to fit bullets standards

TristanFecteau committed:
there was en error in is_market_open

TristanFecteau committed:
valid test + comment for remaining tests

Raphael-Leb committed:
Tests for #84

Raphael-Leb committed:
Finished tests on #84

Raphael-Leb committed:
Update .gitignore

Raphael-Leb committed:
Update .gitignore

TristanFecteau commented on src/test/test_market.py:
```diff
@@ -0,0 +1,78 @@
+            self.assertEqual(True,get_date_in_x_market_days_away(daysaway,startdate)==enddate)
+            logger.info("Return date 2 days away over holiday test successful")
+        finally:
+            pass
```
There should be a newline at the end of the file

TristanFecteau commented on src/bullets/utils/market_utils.py:
```diff
@@ -0,0 +1,45 @@
+    return today
+
+date = datetime(2022, 2, 2, 13, 30, 20)
+print(is_market_open(date, Resolution.MINUTE))
```
What is this? Why are we creating a date value and printing it? This looks like an error.

Also, there should be a newline at the end of the file

Raphael-Leb committed:
Added newlines

Raphael-Leb committed:
Merge branch 'Do-it-here' of https://github.com/AlgoETS/BullETS into Do-it-here

TristanFecteau committed:
fixed a few things

one test has a bad name
there were comments left in the file
variable names did not meet BullETS standards

TristanFecteau committed:
startdate = start_date
